lib/ok.ex

```diff
@@ -231,7 +251,21 @@ defmodule OK do
       end
     end
   end
+  defp nest([{:ok, _} = normal | []]) do
+    # last line of the with block is an ok literal
+    quote do
+      unquote(normal)
+    end
+  end
+  
+  defp nest([{{:., _, [{:__aliases__, _, [:OK]}, :success]}, _, _} = normal | []]) do
+    # last line of the with block is an OK.success macro line
+    quote do
+      unquote(normal)
+    end
+  end
   defp nest([normal | []]) do
+    # last line of the with block is a func
```

Comment:
This comment is used in tandem with the other comments. If you have a variable or literal that is not in the form of the above `{:ok, variable}` or `OK.success variable`, then it will not match in the `case` statement.